### ISSUE TITLE
Sync multiple teams/repos

### ISSUE THREAD

**Issue by zomars** [calcom/linear-to-github#15]
Can we sync the whole organization instead of individual repos? That way we can use Linear as a "central command" for our mix of public and private repos.

**Comment by tedspare** [comment]
Thanks for raising this @zomars. Many-to-many sync is desirable (see #10), agreed. In that case, which Linear team would receive the ticket broadcast by a GH repo?

**Comment by janbaykara** [comment]
I would reframe this feature request as "connect a team to a github org". Our use case is that we want to sync many repos to one linear team.

Ideally, we'd have a way of determining which `Public` linear issues sync to which repo, likely by matching a Linear `team` and `project` to a Github `repo`.

**Comment by PeerRich** [comment]
@janbaykara that needs a bit of architecture discussion as this can quickly creep out of scope

**Comment by janbaykara** [comment]
Agreed. I'm not familiar with the architecture at present. Is there any kind of diagramming / documentation already, or would it just be a case of looking through the code, if I wanted to contribute to that discussion?

**Comment by tedspare** [comment]
@janbaykara I'll share some diagrams and documentation soon™️.

One approach would be to replace the single `Public` label with a label for each repo: `commonknowledge/a`, `commonknowledge/b`, etc

**Comment by janbaykara** [comment]
Yep, a per-repo label would make sense!

**Comment by IronSean** [comment]
I was about to open a new issue when I noticed that this issue also could cover my use case. Here was going to be my proposal:

---

Is there any way to sync one Linear team to different repos? 

*Use Case*
We are a single team that does work on multiple projects that are in multiple repos. We have one linear team because our team has one velocity which might prioritize one project over the other in a given cycle. That means we can't just have different teams per project as the velocity will not be consistent. We use projects to track the different projects in our linear team, and each project has a separate repo and relevant public issues. If we could control which issues go to which repo that would allow us to set up sync for more than one project in the same team.

*Possible solutions*
1. Configuration where a linear project is mapped to a github repo, so `Public` issues on that project get synced to that repo.
2. Separate Linear tags for each repo, so we could customize `ProjectA` to become public on repo A.

**Comment by nikitnikoks** [comment]
It'd be great to have this feature. We also use multiple repos for one team, so we are unable to use the integration at this point.

**Comment by marcklingen** [comment]
Same issue for us; multiple repos (application, sdks, docs) and one team

**Comment by mohab-sameh** [comment]
Same issue here. Came by your issue by coincidence @zomars. Wonder if you found a workaround or if using a monorepo at Cal.com doesn't make a critical issue?

**Comment by galligan** [comment]
Would also support repo-based issue labeling. In dealing with multiple repos, but the same team working on them, this would be a huge improvement.

**Comment by bmhtech07** [comment]
My team could really use this. Small team working across multiple projects in multiple repos (separate tags for each connected repo would be ideal!) 

Is there a timeline for this? Would probably be able to find some time to contribute, but couldn't lead the project.

**Comment by janbaykara** [comment]
I wonder if this is a bit of work and we can usefully try to map out what needs to happen to make this possible?

**Comment by TheMikeyRoss** [comment]
Any update or progress on this one?

**Comment by janbaykara** [comment]
Linear now has a team-based issue sync. https://linear.app/docs/github

**Comment by PeerRich** [comment]
yeah but it lacks a ton of features that synclinear has

**Comment by discdiver** [comment]
Currently, linear allows bi-directional sync for only one repo with its native sync functionality.
